server/service/integration_core_test.go

```diff
@@ -4483,19 +4484,45 @@ func (s *integrationTestSuite) TestQueriesBadRequests() {
 			name:  "Invalid query",
 			query: "",
 		},
+		{
+			tname:    "unsupported platform",
+			name:     "bad query",
+			query:    "select 1",
+			platform: "oops",
+		},
+		{
+			tname:    "unsupported platform",
+			name:     "bad query",
+			query:    "select 1",
+			platform: "charles,darwin",
+		},
+		{
+			tname:    "missing platform comma delimeter",
+			name:     "bad query",
+			query:    "select 1",
+			platform: "linuxdarwin",
+		},
+		{
+			tname:    "missing platform comma delimeter",
+			name:     "bad query",
+			query:    "select 1",
+			platform: "windows darwin",
+		},
 	} {
 		t.Run(tc.tname, func(t *testing.T) {
 			reqQuery := &fleet.QueryPayload{
-				Name:  ptr.String(tc.name),
-				Query: ptr.String(tc.query),
+				Name:     ptr.String(tc.name),
+				Query:    ptr.String(tc.query),
+				Platform: ptr.String(tc.platform),
 			}
 			createQueryResp := createQueryResponse{}
 			s.DoJSON("POST", "/api/latest/fleet/queries", reqQuery, http.StatusBadRequest, &createQueryResp)
 			require.Nil(t, createQueryResp.Query)
 
 			payload := fleet.QueryPayload{
-				Name:  ptr.String(tc.name),
-				Query: ptr.String(tc.query),
+				Name:     ptr.String(tc.name),
+				Query:    ptr.String(tc.query),
+				Platform: ptr.String(tc.platform),
 			}
 			mResp := modifyQueryResponse{}
 			s.DoJSON("PATCH", fmt.Sprintf("/api/latest/fleet/queries/%d", existingQueryID), &payload, http.StatusBadRequest, &mResp)
```

Comment:
it may be worth adding that the error message in the response is what we're expecting.  ie. something like

`assert.Equals(t, mResp.Err, fleet.YourErrorMessage)`

Reply:
Ah nice, sounds good. I'll use `require.ErrorIs` though, per [Lucas's comment](https://github.com/fleetdm/fleet/pull/13923#discussion_r1325884990)